### PR TITLE
Randomizer

### PR DESCRIPTION
**Pull request by vdepasupillagazo**
- created reusable letter generator function
- generate random set of vowels based on requirement parameters
- generate random set of consonants based on requirement parameters
- clean up of unused/duplicate functions
- updating of grid placeholder character to underscore

### CODE REVIEW

File: word-search.py [hunk 1]
```diff
@@ -81,34 +80,91 @@ def print_grid(grid):
...
+# allow letter repetition by default
+def letter_draw(count, letterSet, repeat=True):
+    drawnLetters = ()
+    while len(drawnLetters) < count:
+        index = random.randint(0, len(letterSet)-1)
+        if (not repeat and letterSet[index] in drawnLetters):
+            # choose another letter if already exists
+            continue
+        else:
+            # letters are allowed to repeat
```
Comment (reggyball) [inline word-search.py]: #letters are allowed to repeat or insert first instance of letter

File: word-search.py [hunk 2]
```diff
@@ -81,34 +80,91 @@ def print_grid(grid):
...
+    uConsonantCount = random.randint(0, 2)
+
+    if (uConsonantCount > 0):
+        uConsonantList = letter_draw(uConsonantCount, uncommonConsonants, False)
+        consonantList = letter_draw(
+            consonantCount-uConsonantCount, commonConsonants)
+
+        # insert into random index in cosonantList
+        for uc in uConsonantList:
+            index = random.randint(0, len(consonantList)+1)
```
Comment (reggyball) [inline word-search.py]: @vdepasupillagazo curious what the "+1" is for sa randint and slicing?

Comment (vdepasupillagazo) [reply word-search.py]: Ah, so I double checked and the +1's are not needed indeed. Good catch! :)

File: word-search.py (hunk 1 above)
Comment (vdepasupillagazo) [reply word-search.py]: Comment updated

**Pull request merged**